ENIGMAsystem/SHELL/Universal_System/Extensions/DataStructures/data_structures.cpp

```diff
@@ -50,10 +50,11 @@ template<typename t> bool tequal(t v1, t v2) { return v1 == v2; }
 template<> bool tequal(float v1, float v2)   { return fequal(v1, v2); }
 template<> bool tequal(double v1, double v2) { return fequal(v1, v2); }
 
-namespace enigma {
-  static inline int random_integer(int x) {
-    return int(enigma_user::random(x));
-  }
+template<class RandomIt>
+void mt_random_shuffle(RandomIt first, RandomIt last) {
+  std::random_device rd;
+  std::mt19937 g(rd());
+  std::shuffle(first, last, g);
```

Comment:
It's fine if you want to replace `std::random_shuffle` (deprecated) with `std::shuffle` but this needs to be connected with the user random seed. Review d6fa000a9c4323573703c888bc34e3b1e8780bdf and #1107 to understand why when this eventually fails the CI random test.

Reply:
@JoshDreamland told me to address that in a separate pr and he is in no rush for it to be done.

Reply:
Robert, please steward https://github.com/enigma-dev/enigma-dev/commit/d6fa000a9c4323573703c888bc34e3b1e8780bdf through to submission in another PR. I'm already not a fan of the huge amount of shuffling happening to support BSD; there's no sense blocking this change on fixing already-broken code rather than tweaking it slightly to work on more platforms in some capacity. I agree we should fix it—but not while we're moving earth to support BSD.